### PR TITLE
GSK-2887 Upload tests with project when uploading a suite

### PR DESCRIPTION
**Pull request by kevinmessiaen**
Always send the `project_key` when uploading/downloading tests

### CODE REVIEW

File: tests/communications/test_websocket_actor.py [hunk 1]
```diff
@@ -481,16 +479,14 @@ def do_nothing(row):
 
 
 # FIXME: Internal worker cannot yet load callable due to yaml deserialization issue
```
Comment (Inokinoki) [inline tests/communications/test_websocket_actor.py:481]: idem.

File: tests/test_artifact_download.py [hunk 2]
```diff
@@ -149,17 +150,17 @@ def test_download_callable_function(cf: Artifact):
...
     ],
 )
-def test_download_global_callable_function_from_module(cf: Artifact):
+def test_download_callable_function_from_module(cf: Artifact):
     with MockedClient(mock_all=False) as (client, mr):
         cf.meta.uuid = str(uuid.uuid4())  # Regenerate a UUID to ensure not loading from registry
         cache_dir = get_local_cache_callable_artifact(artifact=cf)
 
         requested_urls = []
         # Prepare global URL
```
Comment (Inokinoki) [inline tests/test_artifact_download.py:159]: Can be changed to prepare URLs, cuz we have no more global artefacts

**Review by Inokinoki — CHANGES_REQUESTED** [review]
LGTM
Just some comments to be removed or polished.

Not related to this PR:
We will neither have global resources anymore, nor "internal worker". So we can also cleanup the `Optional[GiskardClient]`, can we create a card for this?

**Comment by kevinmessiaen** [comment]
> LGTM Just some comments to be removed or polished.
> 
> Not related to this PR: We will neither have global resources anymore, nor "internal worker". So we can also cleanup the `Optional[GiskardClient]`, can we create a card for this?

I've updated th PR and remove the optional client

**Review by Inokinoki — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**